draft-linus-trans-gossip-ct.md

```diff
@@ -179,20 +181,20 @@ corresponding certificate chain for later use in feedback.
 
 When the client later reconnects to any HTTPS server for the same
 domain it again receives a set of SCTs. The client MUST update its
-store of SCTs for the domain and MUST send to the server the ones in
-the store that were not received from that server.
+store of SCTs [should also discard SCTs from unknown logs] for the domain and MUST send to the server the ones in
+the store that were not received from that server [that are for that server].
 
 Note that the SCT store also contains SCTs received in certificates.
 
 The client MUST NOT send the same set of SCTs to the same server more
-often than TBD.
+often than TBD. [suggestion: "sent to the server" only really counts if the server presented a valid SCT in the handshake and the certificate is known to be unrevoked (which will be hard for a MitM to sustain)]
```

Comment:
++ That is true.

Linus: I think he means when a client connections to example.com for the second time (meaning it has some SCTs already) - it "MUST" send the SCTs it has (that are not duplicates.)  But when it updates its store to denote "I sent these SCTs at [some time]" it should only perform that update if the server presents a valid SCT and the certificate is known to be unrevoked.  Those criteria are more difficult to adhere to in the event of a MITM with a mis-issued cert. By not performing the store update if those criteria are not met we increase our gossip frequency  of that data.